src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,30 @@
+package duke.command;
+
+import duke.main.*;
+import duke.exception.DukeException;
+import duke.task.*;
+
+public class DeleteCommand implements Command {
+    private int index;
+
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    public void execute(Storage storage, Ui ui, TaskList tasks) throws DukeException {
+        try {
+            Task deleted = tasks.removeTask(index);
+            ui.output(String.format("Noted. I've removed this task:\n  %s\nNow you have %d tasks in the list",
+                    deleted.toString(), tasks.getTasksSize()));
+            storage.writeToFile(tasks);
+        }
+        catch (Exception e) {
```

Comment:
We should specify which exception we are catching (DukeException or IOException in this case) and not use the base class Exception to catch all exceptions